spec/praxis/api_definition_spec.rb

```diff
@@ -0,0 +1,159 @@
+require 'spec_helper'
+
+describe Praxis::ApiDefinition do
+  
+  subject(:api){  Praxis::ApiDefinition.instance }  
+  
+  # Without getting a fresh new ApiDefinition it is very difficult to test stuff using the Singleton
+  # So for some tests we're gonna create a new instance and work with it to avoid the singleton issues
+  let(:non_singleton_api) do
```

Comment:
maybe there is a way, but the problem is that there are DSL functions that get back to the class instance...I don't know there might be some trickery that can be done. But unless you see something immediate, I'd rather merge this in so that the other things in the queue can trickle in to master too.